### PR TITLE
Handle `null` text values in RankedDocsResults.asMap()

### PR DESCRIPTION
**Pull request by JoanFM**
Small code trying to fix an issue with `asMap` that is not coherent with the requirement of `text` being Nullable.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/results/RankedDocsResultsTests.java [hunk 1]
```diff
@@ -37,6 +38,31 @@ public static RankedDocsResults.RankedDoc createRandomDoc() {
...
+
+        var docWithText = new RankedDocsResults.RankedDoc(index, randomFloat(), "Sample text");
+        var mapWithText = docNullText.asMap();
+        assert mapNullText.containsKey("index");
+        assert mapNullText.containsKey("relevanceScore");
+        assert mapNullText.containsKey("text");
+
+        assert mapNullText.get("index").equals(index);
+        assert mapNullText.get("relevanceScore").equals(score);
+        assert mapNullText.get("text").equals("Sample text");
```
Comment (davidkyle) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/inference/results/RankedDocsResultsTests.java:62]: You can simplify these assertions by using Hamcrest matchers

```suggestion
        var mapNullText = new RankedDocsResults.RankedDoc(index, score, null).asMap();
        assertThat(mapNullText, Matchers.is(Map.of("ranked_doc", Map.of("index", index, "relevance_score", score))));

        var mapWithText = new RankedDocsResults.RankedDoc(index, score, "Sample text").asMap();
        assertThat(mapWithText, Matchers.is(Map.of("ranked_doc", Map.of("index", index, "relevance_score", score, "text", "Sample text"))));
```

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by JoanFM** [comment]
Hey @davidkyle,

how can I check what is wrong with CI?

Thanks,

Joan

**Comment by davidkyle** [comment]
@elasticmachine test this please

**Comment by davidkyle** [comment]
> how can I check what is wrong with CI?

The backwards compatibility upgrade tests failed because there were out of date, the branch need updating

**Pull request merged**